src/System.Text.Json/System/Text/Json/JsonValueType.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Text.Json
+{
+    public enum JsonValueType
+    {
+        Unknown,
+        Object,
+        Array,
+        Number,
+        String,
+        True,
+        False,
+        Null,
+        Undefined,
+        NaN,
```

Comment:
Should we make this an option that the user can toggle? Or the inverse, have a more strict reader that only allows valid JSON?